src/main/java/edu/kpi/testcourse/rest/AuthenticatedApiController.java

```diff
@@ -71,4 +77,21 @@ public AuthenticatedApiController(
       );
     }
   }
+
+  @Get(value = "/urls", processes = MediaType.APPLICATION_JSON)
+  public HttpResponse<String> getAll(
+    @Body UrlShortenRequest request,
+    Principal principal,
+    HttpRequest<?> httpRequest,
+    Logic logic) throws UrlRepository.PermissionDenied {
+    try {
+      var aliasList = logic.getAllAliasesForUser(principal.getName());
```

Comment:
I’m still unsure with those var types in java. If it’s okay, live them, I don’t mind, it’s not a mistake I guess)
But if strict typisation is better, you can change it like this
```suggestion
      List<UrlAlias> aliasList = logic.getAllAliasesForUser(principal.getName());
```